ui.frontend/src/main/webpack/site/js/components/searchResults.ts

```diff
@@ -1,24 +1,20 @@
-import { OnSearchItemClickCallback } from '../types/callbacks'
+import { onSearchItemClick } from '../service/serviceEvent'
 import { buildSearchItem, SearchItem } from './searchItem'
 
 type SearchResultsOptions = {
   searchItems: SearchItem[]
   tabId: string
-  onSearchItemClick?: OnSearchItemClickCallback
 }
 
 export const generateSearchItemList = (
   searchItems: SearchItem[],
-  onSearchItemClick?: OnSearchItemClickCallback,
 ): HTMLDivElement[] => {
   return searchItems.map((searchItem) => {
     const searchItemElement = buildSearchItem(searchItem)
-
+    searchItemElement.addEventListener('SAAS-search-submit', () => {})
     if (onSearchItemClick) {
-      searchItemElement.addEventListener('click', (event) => {
-        event.preventDefault()
-
-        onSearchItemClick?.(searchItem.title)
+      searchItemElement.addEventListener('click', () => {
+        searchItemElement.dispatchEvent(onSearchItemClick)
```

Comment:
Same here could use event.target.dispatchEvent.

Reply:
This is done